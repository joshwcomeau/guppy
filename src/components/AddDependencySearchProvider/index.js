// @flow
export { default } from './AddDependencySearchProvider';
