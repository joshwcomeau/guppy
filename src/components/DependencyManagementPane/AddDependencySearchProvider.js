// @flow
import React, { Component } from 'react';
import { InstantSearch, Configure } from 'react-instantsearch/dom';

import { ALGOLIA_KEYS } from '../../constants';

type Props = {
  children: React$Node,
};

class AddDependencySearchProvider extends Component<Props> {
  render() {
    return (
      <InstantSearch {...ALGOLIA_KEYS}>
        <Configure
          attributesToRetrieve={[
            'name',
            'version',
            'description',
            'modified',
            'humanDownloadsLast30Days',
            'license',
          ]}
          analyticsTags={['guppy', 'guppy-search']}
          hitsPerPage={10}
        />
        {this.props.children}
      </InstantSearch>
    );
  }
}

export default AddDependencySearchProvider;
