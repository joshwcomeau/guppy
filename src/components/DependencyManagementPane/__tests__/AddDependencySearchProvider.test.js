import React from 'react';
import { shallow } from 'enzyme';

import AddDependencySearchProvider from '../AddDependencySearchProvider';

describe('AddDependencySearchProvider component', () => {
  const wrapper = shallow(
    <AddDependencySearchProvider>
      <div />
    </AddDependencySearchProvider>
  );

  it('should render SearchBox', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
