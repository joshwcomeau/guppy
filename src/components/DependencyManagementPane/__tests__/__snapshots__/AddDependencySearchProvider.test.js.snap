// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDependencySearchProvider component should render SearchBox 1`] = `
<CreateInstantSearch
  apiKey="7492903b80561e70bff1359d7052b4ae"
  appId="OFCNCOG2CU"
  indexName="npm-search"
  refresh={false}
  root={
    Object {
      "Root": "div",
      "props": Object {
        "className": "ais-InstantSearch__root",
      },
    }
  }
>
  <AlgoliaConfigure(UnknownComponent)
    analyticsTags={
      Array [
        "guppy",
        "guppy-search",
      ]
    }
    attributesToRetrieve={
      Array [
        "name",
        "version",
        "description",
        "modified",
        "humanDownloadsLast30Days",
        "license",
      ]
    }
    hitsPerPage={10}
  />
  <div />
</CreateInstantSearch>
`;
